main/store/migrations/index.js

```diff
@@ -887,7 +887,11 @@ const migrations = {
     Object.values(initial.main.networks.ethereum).forEach((network) => {
       const { id } = network
       const { name = '', symbol = '' } = nativeCurrencyMap[id] || {}
-      const nativeCurrency = initial.main.networksMeta.ethereum[id].nativeCurrency
+
+      // ensure chain exists in networksMeta
+      initial.main.networksMeta.ethereum[id] = initial.main.networksMeta.ethereum[id] || {}
+
+      const { nativeCurrency } = initial.main.networksMeta.ethereum[id]
```

Comment:
I don't think this is quite right, it still throws an exception below when trying to reference an undefined `nativeCurrency`. I'm updating and adding a couple tests

Reply:
ok, could probably do with a test which handles a chain which has no networksMeta value too

Reply:
yep, there's one in there